user/src/com/google/gwt/user/client/ui/CellPanel.java

```diff
@@ -72,6 +72,8 @@ public abstract class CellPanel extends ComplexPanel {
 
   public CellPanel() {
     table = DOM.createTable();
+    //aria role to indicate it is a plain layout table
+    table.setAttribute("role", "presentation");
```

Comment:
Instead of setting attributes directly, consider using the gwt aria package like other classes do to assign their roles, see TabBar, CellTree, etc.
```suggestion
    Roles.getPresentationRole().set(table);
```

Leaving tests as you have them seems to make sense, we want to verify that the dom is as expected. It could also make sense to query the dom in each of these cases for _all_ tables within the current parent, and validate that each has a role set (at a quick glance, in all cases you've modified we should have the "presentation" role assigned, I don't see nested elements which should actually be presented as tables).